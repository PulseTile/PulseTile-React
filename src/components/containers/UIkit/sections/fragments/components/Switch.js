import React from 'react';

/**
 * This component returns content of Switch section
 */
const Switch = () => {
  return (
    <div id="switch" className="ui-section">
      <strong className="ui-title">Switch</strong>
      <div className="form-group-wrapper">
        <div className="row">
          <div className="col-xs-12 col-sm-6">
            <div className="form-group">
              <label className="control-label">Enabled</label>
              <div className="input-holder">
                <div className="switch-group">
                  <label className="switch">
                    <input type="radio" name="switch1" value="1" checked />
                    <div className="slider">
                      <span className="text text-check-true">Yes</span>
                      <span className="text text-check-false">No</span>
                    </div>
                  </label>
                  <label className="switch">
                    <input type="radio" name="switch1" value="2" />
                    <div className="slider">
                      <span className="text text-check-true">Yes</span>
                      <span className="text text-check-false">No</span>
                    </div>
                  </label>
                </div>
              </div>
            </div>
          </div>
          <div className="col-xs-12 col-sm-6">
            <div className="form-group">
              <label className="control-label">Disabled</label>
                <div className="input-holder">
                  <div className="switch-group">
                    <label className="switch">
                      <input type="radio" name="switch2" value="1" checked disabled />
                      <div className="slider disabled">
                        <span className="text text-check-true">Yes</span>
                        <span className="text text-check-false">No</span>
                      </div>
                    </label>
                    <label className="switch">
                      <input type="radio" name="switch2" value="2" disabled />
                      <div className="slider disabled">
                        <span className="text text-check-true">Yes</span>
                        <span className="text text-check-false">No</span>
                      </div>
                    </label>
                  </div>
                </div>
              </div>
          </div>
        </div>
      </div>
    </div>
  );
};

export default Switch;
