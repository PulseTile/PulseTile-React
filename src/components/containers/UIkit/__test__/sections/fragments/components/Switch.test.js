import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import Switch from '../../../../sections/fragments/components/Switch';

Enzyme.configure({ adapter: new Adapter() });

describe('Component <Switch />', () => {
  it('should renders Switch with props correctly', () => {
    const component = shallow(<Switch />);
    expect(component).toMatchSnapshot();
  });
});
