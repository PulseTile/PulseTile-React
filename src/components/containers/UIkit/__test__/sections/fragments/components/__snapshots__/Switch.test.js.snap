// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component <Switch /> should renders Switch with props correctly 1`] = `
<div
  className="ui-section"
  id="switch"
>
  <strong
    className="ui-title"
  >
    Switch
  </strong>
  <div
    className="form-group-wrapper"
  >
    <div
      className="row"
    >
      <div
        className="col-xs-12 col-sm-6"
      >
        <div
          className="form-group"
        >
          <label
            className="control-label"
          >
            Enabled
          </label>
          <div
            className="input-holder"
          >
            <div
              className="switch-group"
            >
              <label
                className="switch"
              >
                <input
                  checked={true}
                  name="switch1"
                  type="radio"
                  value="1"
                />
                <div
                  className="slider"
                >
                  <span
                    className="text text-check-true"
                  >
                    Yes
                  </span>
                  <span
                    className="text text-check-false"
                  >
                    No
                  </span>
                </div>
              </label>
              <label
                className="switch"
              >
                <input
                  name="switch1"
                  type="radio"
                  value="2"
                />
                <div
                  className="slider"
                >
                  <span
                    className="text text-check-true"
                  >
                    Yes
                  </span>
                  <span
                    className="text text-check-false"
                  >
                    No
                  </span>
                </div>
              </label>
            </div>
          </div>
        </div>
      </div>
      <div
        className="col-xs-12 col-sm-6"
      >
        <div
          className="form-group"
        >
          <label
            className="control-label"
          >
            Disabled
          </label>
          <div
            className="input-holder"
          >
            <div
              className="switch-group"
            >
              <label
                className="switch"
              >
                <input
                  checked={true}
                  disabled={true}
                  name="switch2"
                  type="radio"
                  value="1"
                />
                <div
                  className="slider disabled"
                >
                  <span
                    className="text text-check-true"
                  >
                    Yes
                  </span>
                  <span
                    className="text text-check-false"
                  >
                    No
                  </span>
                </div>
              </label>
              <label
                className="switch"
              >
                <input
                  disabled={true}
                  name="switch2"
                  type="radio"
                  value="2"
                />
                <div
                  className="slider disabled"
                >
                  <span
                    className="text text-check-true"
                  >
                    Yes
                  </span>
                  <span
                    className="text text-check-false"
                  >
                    No
                  </span>
                </div>
              </label>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
